src/lib.rs.in

```diff
@@ -802,6 +802,29 @@ class DocumentAccessResult {
     method DocumentAccessChange::errors(&self) -> FailedResult; alias getErrors;
 });
 
+foreigner_class!(
+    class PolicyCachingConfig{
+        self_type PolicyCachingConfig;
+        constructor policy_caching_config::create(max_entries: usize) -> PolicyCachingConfig;
+    }
+);
+
+foreigner_class!(
+    class IronOxideConfig{
+        self_type IronOxideConfig;
+        constructor ironoxide_config::create(policy_caching: &PolicyCachingConfig, sdk_operation_timeout: Option<&Duration>) -> IronOxideConfig;
+    }
+);
+
+foreigner_class!(
+    class Duration{
```

Comment:
Perhaps we should just use Java's built-in [Duration](https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html) instead of creating our own?

Reply:
I played with rust_swig's `foreign_typemap` macro for a while and couldn't make it work. Unless you can think of an easy way to accomplish this, I think leaving it as our own `Duration` is acceptable.